db/index.js

```diff
@@ -6,17 +6,22 @@ var DB = (() => {
     code = code.toUpperCase()
 
     // Real DB code will go here
-    return db.find(user => user.code === code) || null
+    return db.find(user => user.code === code) || {"error":  'errors.login.code'}
   }
 
   const validateUser = ({ code, sin, dateOfBirth }) => {
     code = code.toUpperCase()
     sin = cleanSIN(sin)
 
     let row = db.find(user => {
-      if (user.code === code && user.sin === sin && user.dateOfBirth === dateOfBirth) return user
+      if (user.sin === sin && user.dateOfBirth === dateOfBirth) {
+        return user
+      }
     })
-
+    if (row && row.code !== code) {
+      // code from the returned user doesn't match
+      return {"error": "errors.login.codeMatch"}
```

Comment:
Yeah, this is what I ran into as well, having to return potentially 3 different values:

- `null`: didn't find anything
- `row`: found a match
- `??`: sin and DoB match a given user but the code is wrong

In line with my earlier comment, I would prefer not to return an error string here. 

In which case, 
 
- we have 3 return values
- we can create another function
- if "validateUser" returns null, we can see if "validateCode" returns a row, and (if it does), we can do a logic check in the route for the session DoB plus the session SIN in `postLogin`.

Reply:
Oh I typed my earlier comment before I saw this

Reply:
👍 This looks perfect, will do

Reply:
Just to be clear, those last three bullet points were "either/or"s. Doing any one of those would work.